src/sentry/utils/snuba.py

```diff
@@ -298,13 +298,67 @@ def transform_aliases_and_query(**kwargs):
     conditions = kwargs['conditions'] or []
     filter_keys = kwargs['filter_keys']
 
+    def parse_selected_column_condition(column):
+        new_column = [column[0]]
+        new_condition = []
+
+        if column[0] == 'if':
+            if_condition = column[1][0]
+            new_if_condition = []
+
+            if isinstance(if_condition, list) and if_condition[0] == 'in':
+                new_if_condition.append(if_condition[0])
+                in_clause = if_condition[1]
+
+                # ["in", [<column name> , <str literal e.g. "tuple">, <list>]]
+                if in_clause[1] != 'tuple' or not isinstance(in_clause[2], list):
+                    # This is an invalid query, raise something
+                    raise
```

Comment:
What are you raising? You could use `sentry.search.utils.InvalidQuery`

Reply:
Yeah I wasn't sure what to raise here

Reply:
Seems like the user has requested an InvalidQuery.

Reply:
This no longer applies, we now do the bare minimum in parsing so that snuba handles the bulk of it.